# Exploring AI & ML Job Trends in the U.S.

## Notebook Version: v1  
**Focus**: Dataset loading and basic structural preview  

This notebook is part of a versioned project exploring trends in AI/ML job postings in the U.S.  
This version focuses on loading the dataset, checking its structure, and identifying surface-level issues.


In [6]:
#importing the necessary libraries
import numpy as np 
import pandas as pd 

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/ai-and-ml-job-listings-usa/ai_ml_jobs_linkedin.csv


## Dataset Overview

- Source: Kaggle – AI and ML Job Listings USA  
- File path: `/kaggle/input/ai-and-ml-job-listings-usa/ai_ml_jobs_linkedin.csv

## Load and Preview Data

Loading the dataset into a DataFrame and preview the structure to understand its basic layout.


In [7]:
# Load the dataset
us_jobs_df = pd.read_csv('/kaggle/input/ai-and-ml-job-listings-usa/ai_ml_jobs_linkedin.csv')

# Create a working copy
jobs_df = us_jobs_df.copy()

In [8]:
# Preview first 2 rows
jobs_df.head(2)


title          location publishedAt  \
0                     AI/ML Engineer      New York, NY  2024-05-29   
1  Software Engineer - AI/ML Systems  Redwood City, CA         NaN   

  companyName                                        description  \
0      Wesper  THE OPPORTUNITY\n\nWesper is looking for a sma...   
1  Snorkel AI  We're on a mission to democratize AI by buildi...   

     applicationsCount contractType   experienceLevel  \
0  Over 200 applicants    Full-time  Mid-Senior level   
1        51 applicants    Full-time       Entry level   

                                 workType                sector  
0  Engineering and Information Technology   Internet Publishing  
1  Engineering and Information Technology  Software Development

In [9]:
# Check dataset shape
print(f"Rows: {jobs_df.shape[0]}, Columns: {jobs_df.shape[1]}")

# Data types and non-null info
jobs_df.info()

Rows: 862, Columns: 10
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 862 entries, 0 to 861
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   title              862 non-null    object
 1   location           862 non-null    object
 2   publishedAt        850 non-null    object
 3   companyName        861 non-null    object
 4   description        862 non-null    object
 5   applicationsCount  862 non-null    object
 6   contractType       862 non-null    object
 7   experienceLevel    862 non-null    object
 8   workType           862 non-null    object
 9   sector             859 non-null    object
dtypes: object(10)
memory usage: 67.5+ KB


In [12]:
# Summary stats for numeric columns
jobs_df.describe()


title       location publishedAt  \
count                         862            862         850   
unique                        450            164         142   
top     Machine Learning Engineer  United States  2024-05-22   
freq                          146            140         136   

                 companyName  \
count                    861   
unique                   519   
top     Unreal Staffing, Inc   
freq                      45   

                                              description  \
count                                                 862   
unique                                                748   
top     Grammarly is excited to offer a remote-first h...   
freq                                                   17   

          applicationsCount contractType   experienceLevel  \
count                   862          862               862   
unique                  145            5                 7   
top     Over 200 applicants    Full-time  Mid-Senior level   
freq                    371          744               403   

                                      workType                sector  
count                                      862                   859  
unique                                      55                   156  
top     Engineering and Information Technology  Software Development  
freq                                       557                   197

## Initial Observations and Notes

- The dataset contains **862 rows** and **10 columns**.
- Some columns such as `companyName`, `publishedAt`, and `sector` contain missing values.
- Columns like `applicationsCount` and `publishedAt` may need data type conversions in the next version.
- No immediate data loading issues were encountered.

## Next Steps

In the upcoming version:
- Handle missing values
- Convert data types (e.g., to datetime)
- Clean or rename columns where necessary
